Tests/Algorithm/Framework/QCAlgorithmFrameworkTests.cs

```diff
@@ -71,6 +71,45 @@ public void SetsInsightGeneratedAndCloseTimes()
             Assert.IsTrue(construction.Insights.All(insight => insight.CloseTimeUtc != default(DateTime)));
         }
 
+        [Test]
+        public void DelistedSecuritiesInsightsTest()
+        {
+            var algorithm = new QCAlgorithm();
+            algorithm.SubscriptionManager.SetDataManager(new DataManagerStub(algorithm));
+            algorithm.Transactions.SetOrderProcessor(new FakeOrderProcessor());
+            algorithm.SetStartDate(2007, 5, 16);
+            algorithm.SetUniverseSelection(new ManualUniverseSelectionModel());
+
+            var alpha = new FakeAlpha();
+            algorithm.SetAlpha(alpha);
+
+            var construction = new FakePortfolioConstruction();
+            algorithm.SetPortfolioConstruction(construction);
+
+            var actualInsights = new List<Insight>();
+            algorithm.InsightsGenerated += (s, e) => actualInsights.AddRange(e.Insights);
+
+            var security = algorithm.AddEquity("SPY", Resolution.Daily);
+            var tick = new Tick
+            {
+                Symbol = security.Symbol,
+                Value = 1,
+                Quantity = 2
+            };
+
+            security.SetMarketPrice(tick);
+            security.IsDelisted = true;
+
+            // Trigger Alpha to emit insight
+            algorithm.OnFrameworkData(new Slice(new DateTime(2000, 01, 01), new List<BaseData>() { tick }));
+
+            // Manually emit insight
+            algorithm.EmitInsights(Insight.Price(Symbols.SPY, TimeSpan.FromDays(1), InsightDirection.Up, .5, .75));
+
+            // Should be zero because security is delisted
+            Assert.AreEqual(0, actualInsights.Count);
```

Comment:
Cool! Maybe we can make `security.IsDelisted = true;` the bool a argument for the test so we assert the count is 2 for when delisted is `false`

Reply:
Gotcha, made it two test cases one for delisted one for not